core/src/main/java/bisq/core/provider/mempool/TxValidator.java

```diff
@@ -56,12 +57,11 @@ public class TxValidator {
 
     private final DaoStateService daoStateService;
     private final FilterManager filterManager;
-    private long blockHeightAtOfferCreation; // Only set for maker.
+    private long feePaymentBlockHeight; // applicable to maker and taker fees
     private final List<String> errorList;
     private final String txId;
     private Coin amount;
-    @Nullable
-    private Boolean isFeeCurrencyBtc = null;
+    private Boolean isFeeCurrencyBtc = true;
```

Comment:
As isFeeCurrencyBtc is only used by one constructor and otherwise is not set (and not used) I think keeping it a nullable Boolean is better. In the use cases where its accessed the constructor is use with a non null value, so no need to initialize it with true.

Reply:
There is one unused constructor as well. Better to remove that.